avocado/core/loader.py

```diff
@@ -308,7 +308,14 @@ def load_test(self, test_factory):
                     if issubclass(obj, test.Test):
                         test_class = obj
                         break
-        test_instance = test_class(**test_parameters)
+        if test_decorator is not None:
+            if isinstance(test_decorator, tuple):
```

Comment:
This is becoming very hackish. `test_factory` should be known format and well documented. Right now, it' s `:param test_factory: a pair of test class and parameters.`. With this change, it can be some other things as well. Or not :)
I don't like the idea of coding the loader to solve problems that are up to the runner. I'd try to keep the loader free of changes.